### ISSUE TITLE
[A11Y] Error message too generic and not listing the errors neither a description of them.

### ISSUE THREAD

**Issue by Devika008** [pkp/pkp-lib#9352]
## Issue Description
// What is the nature of the issue? How does it impact users?

The last step of submitting a new submission is reviewing the details entered. While the user is on this page although the submission system provides the user with error message, it doesn't say which field(s) are missing or incorrect.

## How to reproduce / Current behavior
// Step-by-step instructions explaining how to reproduce the problem. The environment required, explicit identification of the component in question

1. Create a new submission
2. Proceed through the steps
3. If you have missed filling the information when creating a new submission then the system will show error messages

## What application are you using?
 OJS, OMP or OPS version X.X.X
 
 OJS 3.4 

## Stack used 

- Device: Laptop
- OS: MacOS (Monterey)
- Browser: Chrome
- Version: 117

## Testing method
Manual (keyword)

## Action
// Describe action that can be taken to remediate the issue. 

Include detailed list of errors on the top of the page which will clearly indicate the problem with a link which takes you to the section of the review page where the error is. When we do this, we will meet the WCAG Criteria of [3.3.1 Error Identification - A](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html)

## Additional information
// Please add any screenshots, code snippets, or other information we can use to investigate this issue.

![image](https://github.com/pkp/pkp-lib/assets/112014397/c40ba273-b210-4588-a0f5-9905b846633e)

**Comment by Devika008** [comment]
Please find below the proposed solution 

### For 3.4 and 3.5 

1. Its a good idea to guide users towards specific issues of the form with an [error summary](https://design-system.service.gov.uk/components/error-summary/) to help users guide to the specific component. In the design proposed I haven't linked to the error field as doing so would require constant feedback to the user of the error being revised and if that fails to happen then it could be confusing for the user. Hence presented in the design is only a summary text with errors mentioned. If we can update automatically then it would be best to include links to help users jump to the section and then erase the error once its solved.
2. Together with color ( used to indicate error field), an icon to the left of your error message or validation summary will draw attention to the error and also help users who are differently abled visually. When the user scans the form, the icon will stand out and draw the eye to what needs to be fixed.

**Example 1**

![image](https://github.com/pkp/pkp-lib/assets/112014397/9fb73f1f-4ce9-4353-b315-d69a39a0ce88)

**Example 2**

![image](https://github.com/pkp/pkp-lib/assets/112014397/2b955152-5aad-49e4-96b1-412627c472f5)

### For 3.6 and onwards

1. I would aim for **Inline Validation** whenever possible, that is, as soon as the user has finished filling in a field, an indicator should appear nearby if the field contains an error. This type of error message is easily noticeable; moreover, fixing the error immediately after the field has been completed requires the least [interaction cost](https://www.nngroup.com/articles/interaction-cost-definition/) for users: they don’t need to locate it or navigate to the field, nor do they have to switch context from a new field to return to an old field they thought they had completed successfully. I also understand that there will be situations where inline validation won’t be possible and data entered by the user will need to be sent to a server for verification. For these situations the error conditions described above can be used.

**Example 1**

![image](https://github.com/pkp/pkp-lib/assets/112014397/c970f926-5351-4979-ae93-ffb87320226b)
